src/CommunityToolkit.Aspire.Hosting.SqlDatabaseProjects/ConfigureDacDeployOptionsAnnotation.cs

```diff
@@ -0,0 +1,11 @@
+﻿using Microsoft.SqlServer.Dac;
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// Represents a metadata annotation that specifies dacpac deployment options.
+/// </summary>
+/// <param name="ConfigureDeploymentOptions">deployment options</param>
+public record ConfigureDacDeployOptionsAnnotation(Action<DacDeployOptions> ConfigureDeploymentOptions) : IResourceAnnotation
```

Comment:
I'm wondering if this needs to be `public`. It's basically just a mechanism for us to transport the delegate to the place we need it so I guess it could be `internal`. But maybe the Aspire convention is to have annotations always be public. Perhaps @aaronpowell could shed some light on this?

Reply:
Having the record public simplifies testing (we don't have to rely on internals visible attributes). and it's consistent with annotations in the Aspire source, so let's leave it as `public`.

Reply:
👍